src/OrchardCore.Modules/OrchardCore.Media.Azure/Startup.cs

```diff
@@ -33,28 +38,96 @@ public override void ConfigureServices(IServiceCollection services)
             // Only replace default implementation if options are valid.
             var connectionString = _configuration[$"OrchardCore.Media.Azure:{nameof(MediaBlobStorageOptions.ConnectionString)}"];
             var containerName = _configuration[$"OrchardCore.Media.Azure:{nameof(MediaBlobStorageOptions.ContainerName)}"];
-            if (MediaBlobStorageOptionsCheckFilter.CheckOptions(connectionString, containerName, _logger))
+            if (CheckOptions(connectionString, containerName, _logger))
             {
+                // Register a media cache file provider.
+                services.AddSingleton<IMediaFileStoreCacheFileProvider>(serviceProvider =>
+                {
+                    var hostingEnvironment = serviceProvider.GetRequiredService<IHostingEnvironment>();
+
+                    if (String.IsNullOrWhiteSpace(hostingEnvironment.WebRootPath))
+                    {
+                        throw new Exception("The wwwroot folder for serving cache media files is missing.");
+                    }
+
+                    var mediaOptions = serviceProvider.GetRequiredService<IOptions<MediaOptions>>().Value;
+                    var shellOptions = serviceProvider.GetRequiredService<IOptions<ShellOptions>>();
+                    var shellSettings = serviceProvider.GetRequiredService<ShellSettings>();
+                    var logger = serviceProvider.GetRequiredService<ILogger<DefaultMediaFileStoreCacheFileProvider>>();
+
+                    var mediaCachePath = GetMediaCachePath(hostingEnvironment, DefaultMediaFileStoreCacheFileProvider.AssetsCachePath, shellSettings);
+
+                    if (!Directory.Exists(mediaCachePath))
+                    {
+                        Directory.CreateDirectory(mediaCachePath);
+                    }
+
+                    return new DefaultMediaFileStoreCacheFileProvider(logger, mediaOptions.AssetsRequestPath, mediaCachePath);
+                });
+
+                // Replace the default media file provider with the media cache file provider.
+                services.Replace(ServiceDescriptor.Singleton<IMediaFileProvider>(serviceProvider =>
+                    serviceProvider.GetRequiredService<IMediaFileStoreCacheFileProvider>()));
+
+                // Register the media cache file provider as a file store cache provider.
+                services.AddSingleton<IMediaFileStoreCache>(serviceProvider =>
+                    (IMediaFileStoreCache)serviceProvider.GetRequiredService<IMediaFileStoreCacheFileProvider>());
+
```

Comment:
Hmm, maybe here `IMediaFileStoreCacheFileProvider` would need to inherit from `IMediaFileStoreCache` as it inherits from `IMediaFileProvider`, otherwise another feature would be able to register another implementation that doesn't implement `IMediaFileStoreCache`.

Reply:
Yes, you're right. Done